hooks/useLockout.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Gridiron Survivor.
+// Licensed under the MIT License.
+
+import { useEffect, useState } from 'react';
+
+/**
+ * Function to constantly check the current time every hour and set the lockout accordingly.
+ * @returns - The value of lockedOut.
+ */
+const useLockout = (): boolean => {
+  const [lockedOut, setLockedOut] = useState<boolean>(false);
+  useEffect(() => {
+    /**
+     * Checks if the user is locked out from making a pick.
+     */
+    const checkLockout = (): void => {
+      const currentDateAndTime = new Date();
+      const day = currentDateAndTime.getUTCDay();
+      const hours = currentDateAndTime.getUTCHours();
+      if (
+        (day === 5 && hours >= 0) || // Thursday at 9pm EST
+        day > 5 || // Friday and Saturday
+        day === 0 || // Sunday
+        day === 1 || // Monday
+        (day === 2 && hours < 12) // Tuesday at 9am EST
+      ) {
+        setLockedOut(true);
```

Comment:
Looks like we may be updating this state regardless if the value has changed. 

I wonder if a check could be made to ensure we only update state if the value has changed.

```typescript
      const isLockedOut =
        (day === 5 && hours >= 0) || // Thursday at 9pm EST
        day > 5 || // Friday and Saturday
        day === 0 || // Sunday
        day === 1 || // Monday
        (day === 2 && hours < 12); // Tuesday at 9am EST

      if (isLockedOut !== lockedOut) {
        setLockedOut(isLockedOut);
      }
```

Reply:
I apologize that I dont follow. When Ryan made this function, it is to handle the buttons being disabled, etc. During the specific days/times itll keep lockedOut as true which will lock everything down. Then outside of those days/times itll have it set to false which will unlock everything for users to use. And with the interval it checks the time every hour.

Reply:
@braydoncoyer like @alexappleget said, this is simply checking several statements and, if any of them are true, it will change the state. Are you saying you think this is changing the state multiple times? 

Below is a screenshot of what it is achieving; notice the user cannot click and the links are disabled.
![CleanShot 2024-10-14 at 17 31 46@2x](https://github.com/user-attachments/assets/75734955-e515-440b-ae22-7db66034d82c)

Reply:
@ryandotfurrer Thanks for the clarification! 

My original comment could have been more clear. I _think_ that we may set state to a value (let's say `true`) even if the value is already `true`. This seems redundant. It may not actually work like this -- I'd have to pull down and test locally, but perhaps you know the answer off the top of your head!